### PR TITLE
Upgrade to Calcite 1.22.0

### PR DESCRIPTION
**Pull request by eolivelli**
Update to Apache Calcite 1.22.0.

Original patch contributed by @amitvc

### CODE REVIEW

**Comment by eolivelli** [comment]
```
[ERROR] Failures: 
[ERROR]   SimpleJoinTest.testSimpleJoinNoWhere:502 expected:<1> but was:<2>
[ERROR]   SystemTablesJoinTest.testSimpleJoinNoWhere:121 expected:<1> but was:<2>
[ERROR]   SystemTablesTest.testSysTables:278 expected:<5> but was:<24>
[ERROR]   CalcitePlannerTest.dirtySQLPlansTests:224->assertInstanceOf:786 expecting class herddb.model.planner.BindableTableScanOp but found class herddb.model.planner.TableScanOp
[ERROR]   CalcitePlannerTest.joinsPlansTests:329
[ERROR]   CalcitePlannerTest.simplePlansTests:116->assertInstanceOf:786 expecting class herddb.model.planner.BindableTableScanOp but found class herddb.model.planner.TableScanOp
[ERROR] Errors: 
[ERROR]   RawSQLTest.multipleColumnPrimaryKeyTest:1988 » StatementExecution java.lang.Il...
[INFO] 

```

File: pom.xml [hunk 1]
```diff
@@ -541,6 +540,19 @@
             <activation>
                 <activeByDefault>true</activeByDefault>
             </activation>
+            <repositories>
+                <repository>
+                    <id>apache.snapshots</id>
+                    <name>Apache Development Snapshot Repository</name>
+                    <url>https://repository.apache.org/content/repositories/orgapachecalcite-1078</url>
```
Comment (eolivelli) [inline pom.xml]: This is the repository for 1.22.0rc1
They will release 1.22.0 officially by the end of this week

**Comment by eolivelli** [comment]
@dannychan fixed the problem with the UPDATE statement

https://github.com/apache/calcite/pull/1838

**Comment by eolivelli** [comment]
@amitvc  @diegosalvi I have rebased to current master.
Please take a look.
Calcite 1.22.0 has been released so we are good to go with this patch.

The original author is @amitvc we should preserve this information in the commit message.

Thank you @amitvc

**Comment by eolivelli** [comment]
Bad news, we have problems on JDBC driver !

**Comment by eolivelli** [comment]
No,
it looks like Travis is using a bad version of Calcite

**Comment by eolivelli** [comment]
I have cleaned up Travis cache and started again the job

**Comment by eolivelli** [comment]
Failed with an unrelated error on BLink:
```
ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.583 s <<< FAILURE! - in herddb.index.blink.ConcurrentCheckpointBLinkTest
[ERROR] concurrentReadWithManyEmptyNodes(herddb.index.blink.ConcurrentCheckpointBLinkTest)  Time elapsed: 1.49 s  <<< ERROR!
java.util.concurrent.ExecutionException: java.io.UncheckedIOException: failed to search for 18994
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at herddb.index.blink.ConcurrentCheckpointBLinkTest.concurrentReadWithManyEmptyNodes(ConcurrentCheckpointBLinkTest.java:634)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.io.UncheckedIOException: failed to search for 18994
	at herddb.index.blink.BLink.search(BLink.java:604)
	at herddb.index.blink.ConcurrentCheckpointBLinkTest$Reader.call(ConcurrentCheckpointBLinkTest.java:409)
	at herddb.index.blink.ConcurrentCheckpointBLinkTest$Reader.call(ConcurrentCheckpointBLinkTest.java:347)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.IOException: failed to unload 1249
	at herddb.index.blink.BLink$LockAndUnload.unload(BLink.java:3016)
	at herddb.index.blink.BLink$Node.check_key(BLink.java:2217)
	at herddb.index.blink.BLink.search(BLink.java:598)
	... 6 more
Caused by: java.lang.NullPointerException
	at herddb.index.blink.BLink.unload(BLink.java:466)
	at herddb.index.blink.BLink$LockAndUnload.unload(BLink.java:3014)
	... 8 more
```

**Comment by eolivelli** [comment]
Yes, we have legit failures on JDBC tests:
```
GRAVE: NettyChannel{name=unnamed, id=10, socket=[id: 0x474e6391, L:local:127.0.0.1:7000:false - R:local:E:71918cfc] pending 0 msgs}: error java.lang.IllegalArgumentException: call finish() before read operations
java.lang.IllegalArgumentException: call finish() before read operations
	at herddb.utils.DiskArrayList.iterator(DiskArrayList.java:72)
	at herddb.file.FileRecordSet.iterator(FileRecordSet.java:91)
	at herddb.core.SimpleDataScanner.rewind(SimpleDataScanner.java:103)
	at herddb.model.DataScanner.asEnumerator(DataScanner.java:137)
	at herddb.model.DataScanner$1.enumerator(DataScanner.java:128)
	at org.apache.calcite.linq4j.EnumerableDefaults$11$1.moveNext(EnumerableDefaults.java:1904)
	at herddb.model.planner.EnumerableDataScanner.fetchNext(EnumerableDataScanner.java:56)
	at herddb.model.planner.EnumerableDataScanner.next(EnumerableDataScanner.java:66)
	at herddb.model.DataScanner.consume(DataScanner.java:107)
	at herddb.server.ServerSideConnectionPeer.handleOpenScanner(ServerSideConnectionPeer.java:462)
	at herddb.server.ServerSideConnectionPeer.requestReceived(ServerSideConnectionPeer.java:181)
	at herddb.network.netty.NettyChannel.lambda$handlePduRequest$0(NettyChannel.java:96)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:830)
```

**Comment by eolivelli** [comment]
There is an error on backward compatibily tests, I have a fix but I want to reproduce the error in SimpleJoinTest

**Comment by eolivelli** [comment]
@amitvc @diegosalvi  all tests are passing !
We can consider committing this upgrade before releasing 0.15.

**Comment by eolivelli** [comment]
@diegosalvi  @amitvc  I have updated this branch and fixed conflicts

File: herddb-core/src/main/java/herddb/file/FileRecordSet.java [hunk 2]
```diff
@@ -88,7 +87,11 @@ public void write(DataAccessor object, ExtendedDataOutputStream oo) throws IOExc
         if (!writeFinished) {
             throw new IllegalStateException("RecordSet is still in write mode");
         }
-        return buffer.iterator();
+        try {
+            return buffer.iterator();
```
Comment (diegosalvi) [inline herddb-core/src/main/java/herddb/file/FileRecordSet.java]: Why this try and rethrow without any other actions?

Comment (eolivelli) [reply herddb-core/src/main/java/herddb/file/FileRecordSet.java]: I added debug at this point, I will remove the try.

**Comment by danny0405** [comment]
Nice job and thanks for the feedback.

**Pull request merged**